api/src/controllers/users.ts

```diff
@@ -501,4 +502,45 @@ router.post(
 	respond,
 );
 
+const registerSchema = Joi.object<RegisterUserInput>({
+	email: Joi.string().email().required(),
+	password: Joi.string().required(),
+	first_name: Joi.string(),
+	last_name: Joi.string(),
+});
+
+router.post(
+	'/register',
+	checkRateLimit,
+	asyncHandler(async (req, _res, next) => {
+		const { error, value } = registerSchema.validate(req.body);
+		if (error) throw new InvalidPayloadError({ reason: error.message });
+
+		const usersService = new UsersService({ accountability: null, schema: req.schema });
+		await usersService.registerUser(value);
+
+		return next();
+	}),
+	respond,
+);
+
+const verifyRegistrationSchema = Joi.string();
+
+router.get(
+	'/register/verify-email',
```

Comment:
Wondering if this should just be `/register/verify`, similar to other single-worded routes like `/invite/accept`, etc. and also to be in-line with the service method name, SDK command, ...